### ISSUE TITLE
Use Mimosa2 predictively?

### ISSUE THREAD

**Issue by Jalapenobadger** [borenstein-lab/MIMOSA2shiny#4]
Hi, I was recommended to check out Mimosa2 as an alternative to Melonnpan, and I was wondering if it is actually possible to configure Mimosa it to do a similar thing. In Melonnpan, a table of microbial abundances and a table of metabolite abundances are used to first train the model, and then afterwards a new table of similar microbial abundances can be input into the model to predict metabolites. Can this be done with Mimosa2? It seems rather that it is only currently capable of describing the contribution of various microbes to the abundances of metabolites, but not in a predictive way that can then be applied to new data. Is that true or am I missing something?

Also, what are the data requirements for running Mimosa? Will it work with as few as 12 samples or does it require more replicates?

Thanks so much!
-Pete

**Comment by cnoecker** [comment]
Hi Pete,
Thanks for your interest in MIMOSA2! This is an interesting question. You are correct that MIMOSA2 is more focused on data interpretation and hypothesis generation (i.e. what microbial mechanisms can explain metabolite observations), while Melonnpan is more focused on metabolite prediction. For each metabolite, MIMOSA2 fits a simple linear model relating microbial metabolic potential to metabolite measurements. So it is possible to take the MIMOSA2 model results from one dataset and apply them to another dataset to predict metabolites, but I would only expect that to be informative if the two datasets are from very similar sample populations. Any resulting predictions should definitely be considered with a grain of salt.

We also address this question a little bit in the MIMOSA2 FAQ here: https://borenstein-lab.github.io/MIMOSA2shiny/faqs.html#cmpsAlone

Yes, MIMOSA2 will work with as few as 12 samples. You'll be less likely to find significantly predicted metabolites than you would be with a larger sample size, but you'll still be able to see which metabolites are explained better or worse by shifts in microbiome metabolic potential. 

Hope that helps!

**Comment by Jalapenobadger** [comment]
Yes, that is very helpful! Considering we were unable to get Melonnpan to
work due to a lack of variability, I guess it is possible that this is
precisely because Melonnpan depends on community effects on the
metabolites, and so MIMOSA may work very well for the same data since it
doesn't depend at all on this.
Thanks,
-Pete

**Comment by cnoecker** [comment]
Great, hope it is informative for you. Feel free to re-open this issue if you have additional questions.

**Comment by Jalapenobadger** [comment]
Hey Cecilia,

I have had one issue. Trying to run the data through the online version of
Mimosa2 I get the following error: Error: object 'stoichReac' not found
get_species_cmp_scores species network !rxn_param rxn_param F no_spec_param
humann2_param rel_abund_param

I went through and removed all metabolites without KEGG IDs, and I've tried
renorming by rel abundance using the GalaxyTrakr-Humann2 tool. I've
attached the data I'm trying to run if that helps.
